charts/openshift-etcd-backup/templates/pv.yaml

```diff
@@ -0,0 +1,24 @@
+{{- if .Values.persistence.nfs.enabled }}
+apiVersion: v1
+kind: PersistentVolume
```

Comment:
should this be a PVC? directly managing a PV seems non intuitive

Reply:
No, this should not be a PVC, because it directly configures a specific NFS volume. I'm not sure if my recommendation to directly configure the NFS volume in the pod spec under `spec.volumes` is any more intuitive, but in my opinion it would be the right way for static volumes that are specific to a pod.

Reply:
either managing PVs directly feels like creating Pods directly or i don't understand what is going on here

Reply:
it's all gone :D

Reply:
@vmaillot why is it gone? 

A pre-created dedicated backup volume shall be used here. This volume mustn't be provisioned dynamically over a provisioner. The PV is needed in case of a restore, where one doesn't have a working cluster available. It is therefore crucial to know the exact volume mount information so that you can access the backup data on that volume independently.

Reply:
because we can do it directly into the pod!
as @tongpu mentionned [here](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#nfsvolumesource-v1-core) you are
i guess we will have to discuss that subject

Reply:
@hairmare 
> either managing PVs directly feels like creating Pods directly or i don't understand what is going on here

Managing PVs directly doesn't come up too often, but it is a valid use-case for certain things.

@paraenggu 
> A pre-created dedicated backup volume shall be used here. This volume mustn't be provisioned dynamically over a provisioner. The PV is needed in case of a restore, where one doesn't have a working cluster available. It is therefore crucial to know the exact volume mount information so that you can access the backup data on that volume independently.

Question: The statement that the volume mustn't be provisioned dynamically is only valid in case we're using a static NFS mount correct?

The way I recommended to implement this is to to specify the NFS volume mount directly in the pod specification (under `spec.volumes`), without creating a PV at all. In my opinion there are a few benefits to configure it this way.

1. No one can accidentally delete the PV
2. The NFS configuration is directly configured in the Cronjob specification and it's clear that it should only be used for that.

Regarding restoring we would have to discuss the exact restore scenarios, because I can't yet imagine an etcd restore scenario where you have a working Kubernetes API to fetch the PV from.

Reply:
@tongpu 
> Question: The statement that the volume mustn't be provisioned dynamically is only valid in case we're using a static NFS mount correct?

Usually we will probably have an NFS share for the etcd-backup, but it could also be another NAS-like share.

> The way I recommended to implement this is to to specify the NFS volume mount directly in the pod specification (under `spec.volumes`), without creating a PV at all.

As long as we're able to specify the required GIDs that's fine, but aren't we limited then to NFS? @eni23 idea was that with a PV we get the flexibility to use whatever the customer provides.

> because I can't yet imagine an etcd restore scenario where you have a working Kubernetes API to fetch the PV from

Exactly, that's why we need to be able to access (mount) the share outside the cluster.

Reply:
> > Question: The statement that the volume mustn't be provisioned dynamically is only valid in case we're using a static NFS mount correct?
> 
> Usually we will probably have an NFS share for the etcd-backup, but it could also be another NAS-like share. However, it's never preferred to provision the volume dynamically, as we need it outside the cluster as well (where one doesn't know the internal volume path which was created during a dynamic provisioning in advance).

Yes, that part was clear for me.
 
> > The way I recommended to implement this is to to specify the NFS volume mount directly in the pod specification (under `spec.volumes`), without creating a PV at all.
> 
> As long as we're able to specify the [required GIDs](https://docs.openshift.com/container-platform/4.7/storage/persistent_storage/persistent-storage-nfs.html#storage-persistent-storage-nfs-group-ids_persistent-storage-nfs) that's fine,

If this is required to make it work, we need to add the ability to configure the Security Context to the chart, because I believe it's not yet implemented. This is not related to volume mounting at all.

> but aren't we limited then to NFS? @eni23 idea was that with a PV we get the flexibility to use whatever the customer provides.

With the way it was implemented creating the PV as part of the Helm chart we would have also been limited to NFS. The main questions is if we want to create the PV as part of the Helm chart or want to consume a preconfigured PV.

> > because I can't yet imagine an etcd restore scenario where you have a working Kubernetes API to fetch the PV from
> 
> Exactly, that's why we need to be able to access (mount) the share outside the cluster.

Ok, so we're aligned regarding that.

Reply:
> > As long as we're able to specify the [required GIDs](https://docs.openshift.com/container-platform/4.7/storage/persistent_storage/persistent-storage-nfs.html#storage-persistent-storage-nfs-group-ids_persistent-storage-nfs) that's fine,
> 
> If this is required to make it work, we need to add the ability to configure the Security Context to the chart, because I believe it's not yet implemented. This is not related to volume mounting at all.

It is implemented [here](https://github.com/adfinis-sygroup/helm-charts/blob/feat/add-openshift-etcd-backup-chart/charts/openshift-etcd-backup/values.yaml#L65) but should I just then added the the [supplementalGroups](https://docs.openshift.com/container-platform/4.7/storage/persistent_storage/persistent-storage-nfs.html#storage-persistent-storage-nfs-group-ids_persistent-storage-nfs) setup?

Reply:
To be honest I'm not exactly sure what's going to happen when we run privileged and with uid 0 anyway. If this turns out to be a problem we can add it later.